src/comfy_pack/package.py

```diff
@@ -199,13 +199,25 @@ def get_search_url(sha: str) -> str:
 
 def download_file(url: str, dest_path: Path, progress_callback=None):
     """Download file with progress tracking"""
-    if subprocess.call(["curl", "--version"], stdout=subprocess.DEVNULL) == 0:
-        subprocess.check_call(
-            ["curl", "-L", url, "-o", str(dest_path)],
+
+    # prepare auth token from huggingface if possible
+    if (token := os.getenv("HF_TOKEN")) and ("huggingface" in url):
+        bearer = f"Bearer {token}"
+        urllib_request = urllib.request.Request(
+            url, headers={"Authorization": bearer}
         )
-        return True
+        curl_auth = ["-H", f"Authorization: {bearer}"]
+    else:
+        curl_auth = []
+        urllib_request = url
+
     try:
-        with urllib.request.urlopen(url) as response:
+        if shutil.which("curl"):
+            subprocess.check_call(
+                ["curl", "-L", url, *curl_auth, "--fail", "-o", str(dest_path)],
```

Comment:
`--fail` added so curl actually returns a error code so that an exception can be thrown by check_call.
To get the same error handling as with the urllib code, this was moved into the `try`. shutil change was necessary because the subprocess.call would except if curl does not exist and thus never allow to go to the urllib code for fallback.

Reply:
why use curl? it has external dependency

Reply:
Well... it was there, I did not add it, I just made it work properly :D Plus it is (much) faster downloading.